### ISSUE TITLE
[FEATURE] "smart case" string filtering

### ISSUE THREAD

**Issue by brandonbloom** [deref/exo#104]
**Is your feature request related to a problem? Please describe.**

When filtering logs, I _usually_ want to search case insensitive, but sometimes I want to search case-sensitive.

**Describe the solution you'd like**

See "smartcase" in vim:
https://vim.fandom.com/wiki/Searching

I always have this on by default. Basically, if you search for an all-lowercase string, then the search is case insensitive. If there are any uppercase characters, then the search is case sensitive.

**Describe alternatives you've considered**

Providing some UI or search/filter directives for toggling case sensitivity on and off. I think that should probably happen first: some more general work to improve filtering/querying expression.

**Comment by kendru** [comment]
I'm not sure if the smart casing is a great option, since it's not especially intuitive. I'd rather expose an "advanced" regex filter interface if you need case sensitivity or a more advanced pattern. Eventually, I would love to implement a fuzzy search that functions more like VSCode's command bar. Thoughts?

**Comment by brandonbloom** [comment]
I agree that some kind of UI for "advanced" filtering is probably the right first step before specific functionality such as ignore or smart case.
